Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py

```diff
@@ -90,20 +91,57 @@ def editable(cls, branch, repository=None):
         return True
 
     @classmethod
-    def branch_point(cls, repository, limit=None):
-        cnt = 0
-        commit = repository.commit(include_log=False, include_identifier=False)
-        while cls.editable(commit.branch, repository=repository):
-            cnt += 1
-            if limit and cnt > limit:
-                return None
-            commit = repository.find(argument='HEAD~{}'.format(cnt), include_log=False, include_identifier=False)
-            if cnt > 1 or commit.branch != repository.branch or cls.editable(commit.branch, repository=repository):
-                log.info('    Found {}...'.format(string_utils.pluralize(cnt, 'commit')))
-            else:
-                log.info('    No commits on editable branch')
+    def branch_point(cls, repository):
+        branches = repository.branches_for(remote=None)
+        production_branches = [
+            'remotes/{}/{}'.format(remote, branch)
+            for remote in repository.source_remotes()
+            for branch in branches[remote] if not repository.dev_branches.match(branch)
+        ]
+
+        head = run(
+            [repository.executable(), 'rev-parse', 'HEAD'],
+            cwd=repository.root_path,
+            capture_output=True,
+            encoding='utf-8',
+        ).stdout.strip()
+
+        partial_bases = set()
+        for shard in [
+            production_branches[cls.MERGE_BASE_SHARD_SIZE * i:cls.MERGE_BASE_SHARD_SIZE * (i + 1)]
+            for i in range(1 + len(production_branches) // cls.MERGE_BASE_SHARD_SIZE)
+        ]:
```

Comment:
Do we not have something like the [`itertools` `grouper` example](https://docs.python.org/3/library/itertools.html#itertools-recipes) anywhere?

Reply:
Not that I'm aware of...I'm open to simpler ways to do this, but this seemed relatively straight forward.

Reply:
Oh, yeah, this is relatively straight forward, I'm just slightly surprised we don't have something generic to make this slightly simpler in-line.